src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -719,5 +724,336 @@ .maxstack  2
+        [InlineData(1, 1)]
+        [InlineData(1, 2)]
+        [InlineData(1, 3)]
+        [InlineData(0, 4)]
+        public void CreateSpanIsNotUsedForSpanOfNonSize1Types_NonFullLength(int start, int length)
+        {
+            var csharp = RuntimeHelpersCreateSpan + @$"
+public class Test
+{{
+    public static System.ReadOnlySpan<char> StaticData => new System.ReadOnlySpan<char>(new char[] {{ 'a', 'b', 'c' }}, {start}, {length});
+}}";
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, TestOptions.ReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.StaticData.get", @$"{{
+  // Code size       29 (0x1d)
+  .maxstack  4
+  IL_0000:  ldc.i4.3
+  IL_0001:  newarr     ""char""
+  IL_0006:  dup
+  IL_0007:  ldc.i4.0
+  IL_0008:  ldc.i4.s   97
+  IL_000a:  stelem.i2
+  IL_000b:  dup
+  IL_000c:  ldc.i4.1
+  IL_000d:  ldc.i4.s   98
+  IL_000f:  stelem.i2
+  IL_0010:  dup
+  IL_0011:  ldc.i4.2
+  IL_0012:  ldc.i4.s   99
+  IL_0014:  stelem.i2
+  IL_0015:  ldc.i4.{start}
+  IL_0016:  ldc.i4.{length}
+  IL_0017:  newobj     ""System.ReadOnlySpan<char>..ctor(char[], int, int)""
+  IL_001c:  ret
+}}");
+        }
+
+        [Fact]
+        public void CreateSpanIsNotUsedForNonComplainTypes()
+        {
+            var csharp = RuntimeHelpersCreateSpan + $@"
+public struct SingleByteField
+{{
+    public byte Value;
+}}
+
+public class Test
+{{
+    public static System.ReadOnlySpan<SingleByteField> StaticData => new SingleByteField[] {{ default, default, default }};
+}}
+";
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, options: TestOptions.UnsafeReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.StaticData.get", @$"{{
+  // Code size       12 (0xc)
+  .maxstack  1
+  IL_0000:  ldc.i4.3
+  IL_0001:  newarr     ""SingleByteField""
+  IL_0006:  call       ""System.ReadOnlySpan<SingleByteField> System.ReadOnlySpan<SingleByteField>.op_Implicit(SingleByteField[])""
+  IL_000b:  ret
+}}");
+        }
+
+        [Fact]
+        public void Alignment_FieldsAreAlignedAndPackedAccordingToType()
+        {
+            string csharp = RuntimeHelpersCreateSpan + @"
+public class Test
+{
+    public static System.ReadOnlySpan<byte> ByteData => new System.ReadOnlySpan<byte>(new byte[] { 1, 2, 3 } );
+    public static System.ReadOnlySpan<short> ShortData => new System.ReadOnlySpan<short>(new short[] { 1, 2, 3 });
+    public static System.ReadOnlySpan<int> IntData => new System.ReadOnlySpan<int>(new int[] { 1, 2, 3 });
+    public static System.ReadOnlySpan<long> LongData => new System.ReadOnlySpan<long>(new long[] { 1, 2, 3 });
+}";
+
+            // All blob RVAs are 8-byte aligned, regardless of the data type.
+            string byteBlobRva = ExecutionConditionUtil.IsWindows ? "I_00002AE0" : "I_00002AE0"; // TODO: fix Unix values
```

Comment:
We can't add new TODO comments in the codebase. For something that needs to be addressed later, we file new issues and put link in code. But this looks like something you intend to fix within this PR.

Reply:
> But this looks like something you intend to fix within this PR.

Yes, I'm using CI to get all the values.